src/main/java/strategy/PurchaseOperation.java

```diff
@@ -0,0 +1,11 @@
+package strategy;
+
+import core.basesyntax.FruitTransaction;
+import core.basesyntax.Storage;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Storage storage) {
+        storage.decreaseQuantity(transaction.getFruit(), transaction.getQuantity());
+    }
```

Comment:
Handling Purchase operation.
Check result balance before saving it in the Storage - it should be positive. Throw `RuntimeException` in case the balance is negative.
Currently, there is no check for the resulting balance after the decrease operation. You should verify that after the purchase, the quantity of the fruit in the storage does not become negative. If it does, throw a `RuntimeException` to indicate an illegal operation.